django/contrib/staticfiles/storage.py

```diff
@@ -166,12 +166,15 @@ def converter(matchobj):
             name_parts = name.split(os.sep)
             # Using posix normpath here to remove duplicates
             url = posixpath.normpath(url)
-            url_parts = url.split('/')
-            parent_level, sub_level = url.count('..'), url.count('/')
-            if url.startswith('/'):
+            # Avoid errors in the edge case that the URLs
+            # is like ../fonts/foo.svg#../fonts/foo
+            url_path, fragment = urldefrag(url)
+            url_parts = url_path.split('/')
+            parent_level, sub_level = url_path.count('..'), url_path.count('/')
```

Comment:
Do you refer the changed code is not intuitive, or the original one? `urldefrag` is used in the same file